apps/dashboard/src/app/(dashboard)/dashboard/connect/ecosystem/[ecosystemId]/hooks/use-add-partner.ts

```diff
@@ -0,0 +1,78 @@
+import { useLoggedInUser } from "@3rdweb-sdk/react/hooks/useLoggedInUser";
+import {
+  type UseMutationOptions,
+  useMutation,
+  useQueryClient,
+} from "@tanstack/react-query";
+import type { Ecosystem, Partner } from "../../types";
+
+export type AddPartnerParams = {
+  ecosystem: Ecosystem;
+  name: string;
+  allowlistedDomains: string[];
+  allowlistedBundleIds: string[];
+  permissions: "PROMPT_USER_V1" | "FULL_CONTROL_V1";
+};
+
+export function useAddPartner(
+  options?: Omit<
+    UseMutationOptions<Partner, unknown, AddPartnerParams>,
+    "mutationFn"
+  >,
+) {
+  const { isLoggedIn } = useLoggedInUser();
+  const queryClient = useQueryClient();
+
+  const { mutateAsync: addPartner, isLoading } = useMutation({
+    // Returns the created partner object
+    mutationFn: async (params: AddPartnerParams): Promise<Partner> => {
+      if (!isLoggedIn) {
+        throw new Error("Please login to add a partner");
+      }
+
+      const res = await fetch(
+        `${params.ecosystem.url}/${params.ecosystem.id}/partner`,
+        {
+          method: "POST",
+          credentials: "include",
+          headers: { "Content-Type": "application/json" },
+          body: JSON.stringify({
+            name: params.name,
+            allowlistedDomains: params.allowlistedDomains,
+            allowlistedBundleIds: params.allowlistedBundleIds,
+            permissions: params.permissions,
+          }),
+        },
+      );
+
+      if (!res.ok) {
+        const body = await res.json();
+        console.error(body);
+        if (res.status === 401) {
+          throw new Error(
+            "You're not authorized to add a partner to this ecosystem",
+          );
+        }
+        if (res.status === 406) {
+          throw new Error(
+            "Please setup billing on your account to continue managing this ecosystem",
+          );
+        }
+        throw new Error("Failed to add ecosystem partner");
+      }
+
+      const data = await res.json();
+
+      return data;
+    },
+    onSuccess: (partner, variables, context) => {
+      return Promise.all([
+        options?.onSuccess?.(partner, variables, context),
+        queryClient.invalidateQueries(["partners", variables.ecosystem.id]),
+      ]);
+    },
+    ...options,
```

Comment:
this is kinda funny - if you pass `onSuccess` to `options` here:

- you'd *think* this would be called, but
- because `onSuccess` is in `options` it will absolutely override the pre-defined `onSuccess`

--- 
your options to fix this are:
1. grab "onSuccess" out of options first before spreading options here (rest params spread)
2. put `onSuccess` after the `...options` spread